osfoffline/polling_osf_manager/osf_query.py

```diff
@@ -242,34 +251,39 @@ def _delete_file_folder(self, remote_file_folder):
         resp.close()
 
     @asyncio.coroutine
-    def make_request(self, url, method=None, params=None, expects=None, get_json=False, timeout=180, data=None):
+    def make_request(self, url, method=None, params=None, expects=None, get_json=False, timeout=180, data=None, headers=None, local_file=None, locks=None):
         if method is None:
             method = 'GET'
 
         request = self.request_session.request(
             url=url,
             method=method.upper(),
+            headers=headers,
             params=params,
             data=data
         )
         try:
-            response = yield from asyncio.wait_for(request, timeout)
-        except (
-                aiohttp.errors.ClientTimeoutError, aiohttp.errors.ClientConnectionError,
+            if locks and local_file:
+                asyncio.async(check_for_changes(local_file, locks))
```

Comment:
As discussed yesterday, `asyncio.async` is deprecated. Use `asyncio.ensure_future` instead.
https://docs.python.org/3/library/asyncio-task.html#asyncio.async

Additionally, it appears that `ensure_future` would return a Task object which can be canceled explicitly- meaning that `check_for_changes` could be canceled/terminated without need for the second lock. Is that correct?